backend/src/bundles/auth/auth.service.ts

```diff
@@ -67,9 +73,47 @@ class AuthService {
             });
         }
 
+        const inviterUser = await this.userService.findWithUserDetailsJoined({
+            referralCode,
+        });
+
+        const isReferralCodeValid = referralCode !== 'null';
```

Comment:
Maybe there is a way to avoid such comparison? We could try to add validation schema for query for example